modules/tinymce/src/core/main/ts/content/SetContentImpl.ts

```diff
@@ -98,30 +111,48 @@ const setContentTree = (editor: Editor, body: HTMLElement, content: AstNode, arg
   args.content = isWsPreserveElement(SugarElement.fromDom(body)) ? html : Tools.trim(html);
   setEditorHtml(editor, args.content, args.no_selection);
 
-  if (!args.no_events) {
-    editor.fire('SetContent', args);
-  }
-
   return content;
 };
 
+const getFormatter = (format: string, formatCell: Cell<Record<string, SetContentFormatter>>) => {
+  return Optional.from(formatCell.get()[format]).fold(
+    () => {
+      throw new Error(`Content formatter ${format} not recognized.`);
+    },
+    Fun.identity
+  );
+};
+
 const setupArgs = (args: Partial<SetContentArgs>, content: Content): SetContentArgs => ({
   format: defaultFormat,
   ...args,
   set: true,
   content: isTreeNode(content) ? '' : content
 });
 
-export const setContentInternal = (editor: Editor, content: Content, args: Partial<SetContentArgs>): Content => {
+const setContentInternal = (editor: Editor, content: Content, args: Partial<SetContentArgs>, formatCell: Cell<Record<string, SetContentFormatter>>): Content => {
+  const formatter = getFormatter(args.format || defaultFormat, formatCell);
```

Comment:
```suggestion
  const formatter = getFormatter(args.format ?? defaultFormat, formatCell);
```
This way it only falls back if it's null or undefined and not just falsy.